Mac arm64 binaries are bad because the pipeline installs x64 .NET

Opening a separate issue to track this, after discussion in comments of #56.

The root cause is [a bug in the .NET install task](https://github.com/microsoft/azure-pipelines-tasks/issues/19174) that the Azure DevOps build pipeline uses. We should be able to work around that by using a different method to install .NET, such as the [install shell script](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script).

And this is the error I get when I try to build a csharp library with the NodeApi packages:

```
MSBuild version 17.3.2+561848881 for .NET
  Determining projects to restore...
  Restored /Users/spacepluk/Hacking/crypthulhu/ethdmx/express_server_postgres_testing/HeroGenJs/HeroGenJs.csproj (in 97 ms).
/Users/spacepluk/.dotnet/sdk/6.0.417/Microsoft.Common.CurrentVersion.targets(2302,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi/0.4.26/lib/net6.0/Microsoft.JavaScript.NodeApi.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [/Users/spacepluk/Hacking/crypthulhu/ethdmx/express_server_postgres_testing/HeroGenJs/HeroGenJs.csproj]
/Users/spacepluk/.dotnet/sdk/6.0.417/Microsoft.Common.CurrentVersion.targets(2302,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/lib/net6.0/Microsoft.JavaScript.NodeApi.Generator.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [/Users/spacepluk/Hacking/crypthulhu/ethdmx/express_server_postgres_testing/HeroGenJs/HeroGenJs.csproj]
CSC : warning CS8034: Unable to load Analyzer assembly /Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll : Could not load file or assembly 'Microsoft.JavaScript.NodeApi.Generator, Version=0.4.0.0, Culture=neutral, PublicKeyToken=null, processorArchitecture=AMD64'. The given assembly name or codebase was invalid. (0x80131047) [/Users/spacepluk/Hacking/crypthulhu/ethdmx/express_server_postgres_testing/HeroGenJs/HeroGenJs.csproj]
  HeroGenJs -> /Users/spacepluk/Hacking/crypthulhu/ethdmx/express_server_postgres_testing/HeroGenJs/bin/Debug/net6.0/HeroGenJs.dll
  Unhandled exception. System.BadImageFormatException: Could not load file or assembly '/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll'. An attempt was made to load a program with an incorrect format.

  File name: '/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll'
  /var/folders/__/x_hjp9c93cg4ch1jbvsykm240000gn/T/MSBuildTempspacepluk/tmp8cf02ea4e42a4ab4af5dc1acb226c7b6.exec.cmd: line 2: 72989 Abort trap: 6           dotnet "/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/build/../analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll" "@obj/Debug/net6.0/Microsoft.JavaScript.NodeApi.Generator.rsp"
/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/build/Microsoft.JavaScript.NodeApi.Generator.targets(47,5): error MSB3073: The command "dotnet "/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/build/../analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll" "@obj/Debug/net6.0/Microsoft.JavaScript.NodeApi.Generator.rsp"" exited with code 134. [/Users/spacepluk/Hacking/crypthulhu/ethdmx/express_server_postgres_testing/HeroGenJs/HeroGenJs.csproj]

Build FAILED.

/Users/spacepluk/.dotnet/sdk/6.0.417/Microsoft.Common.CurrentVersion.targets(2302,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi/0.4.26/lib/net6.0/Microsoft.JavaScript.NodeApi.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [/Users/spacepluk/Hacking/crypthulhu/ethdmx/express_server_postgres_testing/HeroGenJs/HeroGenJs.csproj]
/Users/spacepluk/.dotnet/sdk/6.0.417/Microsoft.Common.CurrentVersion.targets(2302,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/lib/net6.0/Microsoft.JavaScript.NodeApi.Generator.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [/Users/spacepluk/Hacking/crypthulhu/ethdmx/express_server_postgres_testing/HeroGenJs/HeroGenJs.csproj]
CSC : warning CS8034: Unable to load Analyzer assembly /Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll : Could not load file or assembly 'Microsoft.JavaScript.NodeApi.Generator, Version=0.4.0.0, Culture=neutral, PublicKeyToken=null, processorArchitecture=AMD64'. The given assembly name or codebase was invalid. (0x80131047) [/Users/spacepluk/Hacking/crypthulhu/ethdmx/express_server_postgres_testing/HeroGenJs/HeroGenJs.csproj]
/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/build/Microsoft.JavaScript.NodeApi.Generator.targets(47,5): error MSB3073: The command "dotnet "/Users/spacepluk/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.26/build/../analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll" "@obj/Debug/net6.0/Microsoft.JavaScript.NodeApi.Generator.rsp"" exited with code 134. [/Users/spacepluk/Hacking/crypthulhu/ethdmx/express_server_postgres_testing/HeroGenJs/HeroGenJs.csproj]
    3 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.42

```

@jasongin  @vmoroz  Thanks for fixing the build, I tried using version 04.34 on mac osx ARM64, but got the below error.

```
node-cblite -> /Users/sumeetsingh/development/ngtos/node-cblite/bin/Release/net6.0/osx-arm64/node-cblite.dll
  Unhandled exception. System.BadImageFormatException: Could not load file or assembly '/Users/sumeetsingh/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.34/analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll'. An attempt was made to load a program with an incorrect format.
  
  File name: '/Users/sumeetsingh/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.34/analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll'
  /var/folders/wj/yl2ysnj11ybbgjfl_qwdr2ym0000gq/T/MSBuildTempsumeetsingh/tmp2abb273f6bce42f992b445886b379fdd.exec.cmd: line 2: 27522 Abort trap: 6           dotnet "/Users/sumeetsingh/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.34/build/../analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll" "@obj/Release/net6.0/osx-arm64/Microsoft.JavaScript.NodeApi.Generator.rsp"
/Users/sumeetsingh/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.34/build/Microsoft.JavaScript.NodeApi.Generator.targets(47,5): error MSB3073: The command "dotnet "/Users/sumeetsingh/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.34/build/../analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll" "@obj/Release/net6.0/osx-arm64/Microsoft.JavaScript.NodeApi.Generator.rsp"" exited with code 134. [/Users/sumeetsingh/development/ngtos/node-cblite/node-cblite.csproj::TargetFramework=net6.0]
```

> @jasongin @vmoroz Thanks for fixing the build, I tried using version 04.34 on mac osx ARM64, but got the below error.
> 
> ```
> node-cblite -> /Users/sumeetsingh/development/ngtos/node-cblite/bin/Release/net6.0/osx-arm64/node-cblite.dll
>   Unhandled exception. System.BadImageFormatException: Could not load file or assembly '/Users/sumeetsingh/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.34/analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll'. An attempt was made to load a program with an incorrect format.
>   
>   File name: '/Users/sumeetsingh/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.34/analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll'
>   /var/folders/wj/yl2ysnj11ybbgjfl_qwdr2ym0000gq/T/MSBuildTempsumeetsingh/tmp2abb273f6bce42f992b445886b379fdd.exec.cmd: line 2: 27522 Abort trap: 6           dotnet "/Users/sumeetsingh/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.34/build/../analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll" "@obj/Release/net6.0/osx-arm64/Microsoft.JavaScript.NodeApi.Generator.rsp"
> /Users/sumeetsingh/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.34/build/Microsoft.JavaScript.NodeApi.Generator.targets(47,5): error MSB3073: The command "dotnet "/Users/sumeetsingh/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.34/build/../analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll" "@obj/Release/net6.0/osx-arm64/Microsoft.JavaScript.NodeApi.Generator.rsp"" exited with code 134. [/Users/sumeetsingh/development/ngtos/node-cblite/node-cblite.csproj::TargetFramework=net6.0]
> ```

It looks like related to the missing dlls for other architectures on NuGet. If you reference the local dlls of the NPM package, you migh get further.

--- a/src/NodeApi.DotNetHost/NodeApi.DotNetHost.csproj
+++ b/src/NodeApi.DotNetHost/NodeApi.DotNetHost.csproj
@@ -5,6 +5,7 @@
     <RootNamespace>Microsoft.JavaScript.NodeApi.DotNetHost</RootNamespace>
     <IsPackable>true</IsPackable>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <PlatformTarget>AnyCPU</PlatformTarget>
   </PropertyGroup>
 
   <ItemGroup>

--- a/src/NodeApi.Generator/NodeApi.Generator.csproj
+++ b/src/NodeApi.Generator/NodeApi.Generator.csproj
@@ -48,6 +48,7 @@
 
   <PropertyGroup>
     <GetTargetPathDependsOn>$(GetTargetPathDependsOn);GetDependencyTargetPaths</GetTargetPathDependsOn>
+    <PlatformTarget>AnyCPU</PlatformTarget>
   </PropertyGroup>
 
   <Target Name="GetDependencyTargetPaths">

--- a/src/NodeApi/NodeApi.csproj
+++ b/src/NodeApi/NodeApi.csproj
@@ -7,6 +7,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
     <SelfContained>false</SelfContained>
+    <PlatformTarget>AnyCPU</PlatformTarget>
   </PropertyGroup>
 
   <PropertyGroup Condition=" '$(PublishAot)' == 'true' ">

